test/manifolds/generalized_torus.jl

```diff
@@ -0,0 +1,122 @@
+using Revise
+using Manifolds
+using GLMakie, Makie
+using OrdinaryDiffEq
+using Test
+
+using StaticArrays
+using DiffEqCallbacks
+using SciMLBase
+using RecursiveArrayTools
+using Manifolds: TFVector
+
+@testset "Torus in ℝ³" begin
+    M = Manifolds.TorusInR3(3, 2)
+    A = Manifolds.DefaultTorusAtlas()
+
+    #p0x = [0.5, -1.2]
+    p0x = [pi / 2, 0.0]
+    X_p0x = [-1.2, 0.4]
+    p = [Manifolds._torus_param(M, p0x...)...]
+    i_p0x = Manifolds.get_chart_index(M, A, p)
+    B = induced_basis(M, A, i_p0x)
+    X = get_vector(M, p, X_p0x, B)
+    @test get_coordinates(M, p, X, B) ≈ X_p0x
+
+    @test_broken norm(X) ≈ norm(M, p0x, TFVector(X_p0x, B))
```

Comment:
@kellertuer Could you check if this test actually makes sense? This is supposed to test that the vector `X` in embedding of the torus has the same norm as calculated using its coordinates in the induced basis and the metric tensor. I'm not sure where I could have a mistake.

Reply:
Does this one error? If so – by how much?
Also just `norm(X)`is a little surprising, is that already in the embedding?

Reply:
Yes, `X` is in the embedding and the difference is large:
```
  Expression: norm(X) ≈ norm(M, p0x, TFVector(X_p0x, B))
   Evaluated: 2.6832815729997477 ≈ 1.4422205101855958
```
I'd expect the embedding to be isometric because the metric is actually defined through that embedding.

Reply:
yes then I would expect that as well. But for the next few days I do not have time to check the code too much I fear.

Reply:
No problem, it can wait.

Reply:
First idea, did you check that the basis is an ONB? Otherwise you might need two bases (one for decomposition one for composition) to cancel out the non-normal-basis vectors?

Reply:
`B` is not ONB but why would it matter here?

Reply:
`get_vector` and `get_coordinates` are inverses of each other, that's one thing that I've checked.

Reply:
Hm, ok; then I would have to check how you compute the norm on the right (because then it can not be get_vector&norm in embedding, right? Because if they are inverses that should be the same as X).

Reply:
Yes, there is likely an error in both `get_vector` and `get_coordinates`. I've noticed that I have mixed up notation in a few places so anyway it's better if you wait until I fix that.

Reply:
Ok, but I hope that in the best case the code works fine after you fixed that 🤓

Reply:
I've found the errors but one of them looks like an issue with API. Currently `local_metric` expects point in the embedding while arguably it would make sense to pass there coordinate in a basis. It would be nice to be able to tell these two cases apart by method signatures. Do you have an idea for that?

Reply:
Cool, that you found it 

Which `local_metric`do you refer to? The one on the sphere
https://github.com/JuliaManifolds/Manifolds.jl/blob/71d9d8137e6418b4931581ab083da0cc370ac621/src/manifolds/Sphere.jl#L285-L291
Does what I would expect – the point p on the manifold (or would you prefer _that_ in charts/coordinates?) 

Because there `G = local_metric(M, p,B)` returns the metric as a matrix such that for tangent vectors given in coordinates with respect to `B` we have `c1'*G*c2` which is what I would expect.

One could write one that is `g = local_metric(M,p)` such that `X'*g*Y` works for vectors (not coordinates).

So I am not sure what you mean by "point in a basis", since we do not have a basis on the manifold.

Reply:
Sorry, it was a bit late -- I meant passing point as parameters in a chart. So probably something like `local_metric(M::AbstractManifold, A::AbstractAtlas, i, a)` for coefficients of the metric tensor at point with parametrization `a` in chart `i` from atlas `A` in basis induced by that chart at that point.

What I'd like to develop here a bit is working on a manifold without an embedding.

Reply:
Ah, but then `A` solves your thing here :) I like the signature you propose.

What I also meant was, that there is a (though only subtle and small) difference between point-representation and embedding. But sure in charts is what you meant.